File: backend/src/main/java/com/ramennsama/springboot/lms/service/ProgressService.java
package com.ramennsama.springboot.lms.service;

import com.ramennsama.springboot.lms.dto.response.CourseProgressResponse;

public interface ProgressService {
    void markLessonAsCompleted(Long lessonId);
    CourseProgressResponse getCourseProgress(Long courseId);
}

File: frontend/src/services/progressService.js
import api from './api';

export const progressService = {
  completeLesson: async (lessonId) => {
    return api.post(`/progress/${lessonId}/complete`);
  },
  getCourseProgress: async (courseId) => {
    return api.get(`/progress/course/${courseId}`);
  }
};
